### PR TITLE
feat(full_url_for): utilize hexo-util module

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do? 

https://github.com/hexojs/hexo-util/pull/84

Returns a url with the `config.url` prefixed.

**Use Case 1**

`link[canonical]` tag, which is one of a best practice for SEO, needs an absolute url of current page.

According to [the test carried out by theme NexT team](https://github.com/theme-next/hexo-theme-next/pull/1143#issuecomment-526828133):

> `{{ page.permalink }}` works on every page exclude index and archive page.
> `{{ url }}` is everywhere.

A `full_url_for(page.path)` can be used in this case.

**Use Case 2**

Use url to share the post at social media, which is similar to the first case:

- Sina Weibo `http://service.weibo.com/share/share.php?appkey=&title=[title]&url=[url]&pic=[site logo]&searchPic=false&style=simple`
- Twitter `https://twitter.com/intent/tweet?text=[title]&url=[url]`
- Facebook `https://www.facebook.com/sharer/sharer.php?u=[url]`
- Telegram `https://t.me/share/url?url=[url]&text=[title]`

Absolute urls are needed for `[url]` and `[site logo]`, and `full_url_for(page.path)` & `full_url_for(${site_logo_url})` can be used.

**Use Case 3**

All examples of [structured data](https://developers.google.com/search/docs/guides/intro-structured-data) (Image, Page path...) and [open-graph](https://ogp.me/) use absolute urls. If they are best practices, then `full_url_for` helper can be used.

## How to test

```sh
git clone -b full_url_for https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Screenshots

I have implemented a [`_full_url_for`](https://github.com/SukkaW/hexo-theme-suka/blob/529b05e723365722861661eae23dd98d5870973d/includes/helpers/url_for.js) for my suka theme and it works perfectly.

## Pull request tasks

- [x] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

**Comment by SukkaW** [comment]
This feature could be released with Hexo 4.1.0.

**Comment by SukkaW** [comment]
This might be a alternate solution for #3604

**Comment by SukkaW** [comment]
Test will be passed after hexo's dependencies updates.

File: lib/plugins/helper/full_url_for.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+const { full_url_for } = require('hexo-util');
+
+module.exports = function(path = '/') {
```
Comment (curbengh) [inline lib/plugins/helper/full_url_for.js]: can be just `path`, hexo-util.full_url_for already set default parameter.

**Comment by curbengh** [comment]
> Test will be passed after hexo's dependencies updates.

Just update hexo-util to ^1.2.0.

File: lib/plugins/helper/full_url_for.js (hunk 1 above)
Comment (SukkaW) [reply lib/plugins/helper/full_url_for.js]: Fixed, and related test has been added.

**Pull request merged**